game/src/game_core/event.rs

```diff
@@ -136,47 +114,106 @@ impl Event {
             popup_type
         );
     }
+
     /// Check if event is still active
     pub fn is_active(&self) -> bool {
         // check if time since event creation is greater than the duration of the event
-        self.start_time.elapsed().unwrap() < self.duration
+        !self.duration <= 0
     }
+
+    /// Triggers the event and activates its effect
+    /// # Arguments
+    /// * `restore` - If true the event will be deactivated and the resources will be restored
+    /// * `gamestate` - The gamestate which is used to access the player and the machines
+    /// * `sender` - The sender which is used to send the popup to the screen
+    pub fn action(&self, restore: bool, gamestate: &mut GameState, sender: &Sender<StackCommand>) {
+        const KOMETENEINSCHLAG_NAME: &str = KOMETENEINSCHLAG[0];
```

Comment:
Wäre ein Kommentar nicht besser als neue Const ? xD

Reply:
Das ist so gelöst, weil ich in dem match nicht auf KOMETENEINSCHLAG[0] zugreifen kann (bzw. KOMETENEINSCHLAG)

Reply:
Top passt